### PR TITLE
added changes in deberta dropout

### PR DESCRIPTION
**Pull request by avacaondata**
# What does this PR do?

This PR does the foillowing. 
1- Add a task dropout to deberta config, that is, cls_dropout. This is similar to what Bert and others have. This is justified since in Deberta paper adjusting task dropout is very important, therefore I think it shouldn't be just be set to the hidden_dropout_prob. While the latter should, in general, not be changed, cls_dropout should be tuned to obtain best results, as stated in: https://arxiv.org/abs/2006.03654
2. Armonize the use of Dropout. In general, Deberta uses Stable Dropout, which was also used in Classification. However, I don't understand why, on Token Classification normal Dropout was used. In some experiments I carried out, StableDropout worked better than normal dropout also for token classification, and it is the type of dropout that is used in all other Deberta layers, therefore I think it is better to have the same type of dropout in all task layers. 
I also changed a little bit how the dropout is created to remove redundant lines, but as you see there are only a few changes, they're not many lines of code, but for clarity and simplicity I decided to change that also; e.g (instead of first getting cls_dropout if existing and then if it is None take hidden_dropout_prob, I do it all in one line in getattr).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR

@BigBird01 @LysandreJik @patrickvonplaten

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
Hey, thanks for the changes @alexvaca0!

Regarding the first change, for backwards compatibility purposes, I would keep `cls_dropout`'s value to `None` so that the line `getattr(config, "cls_dropout", self.config.hidden_dropout_prob)` returns the value of `self.config.hidden_dropout_prob` if `cls_dropout` is not passed.

Regarding the second change, from `nn.Dropout` to `StableDropout`, I would like to hear from @BigBird01 to know if this change is warranted, given that it was part of the initial implementation.

**Comment by maziyarpanahi** [comment]
Hi,

Just as an FYI, changing `nn.Dropout` to `StableDropout` will stop the models supporting `saved_model` for TensorFlow when trying to save them on disk for TensorFlow serving. This is already an issue in `TFDebertaV2ForSequenceClassification` which both `nn.Dropout` and `StableDropout` were tested on 4 different datasets for multi-class classification and the results were pretty much the same between these two. (I already changed it to nn.Dropout so I can save them as saved_model, the performance in production is exactly the same as StableDropout at least for me)

reference: https://github.com/huggingface/transformers/issues/16484
